waspc/src/Wasp/Generator/Job/Process.hs

```diff
@@ -94,15 +91,30 @@ runProcessAsJob process jobType chan =
         else P.interruptProcessGroupOf processHandle
       return $ ExitFailure 1
 
-runNodeCommandAsJob :: Path' Abs (Dir a) -> String -> [String] -> J.JobType -> J.Job
-runNodeCommandAsJob fromDir command args jobType chan = do
+-- | Decodes given byte string while assuming it is using locale encoding (which is system's default encoding).
+decodeLocaleEncoding :: ByteString -> Text
+decodeLocaleEncoding = decodeEncoding localeEncoding
+
+-- | Decodes given byte string while assuming it is using provided text encoding.
+decodeEncoding :: TextEncoding -> ByteString -> Text
+decodeEncoding enc
+  | textEncodingName enc == textEncodingName latin1 = decodeLatin1
+  -- This will replace any invalid characters with \xfffd.
+  | textEncodingName enc == textEncodingName utf8 = decodeUtf8With onErrorUseReplacementChar
+  | otherwise = error $ "Encoding " ++ textEncodingName localeEncoding ++ " is not supported."
+  where
+    onErrorUseReplacementChar _ _ = Just '\xfffd'
+
+runCommandThatRequiresNodeAsJob :: Path' Abs (Dir a) -> String -> [String] -> J.JobType -> J.Job
+runCommandThatRequiresNodeAsJob fromDir command args jobType chan = do
   errorOrNodeVersion <- getNodeVersion
   case errorOrNodeVersion of
     Left errorMsg -> exitWithError (ExitFailure 1) (T.pack errorMsg)
     Right nodeVersion ->
       if SV.isVersionInRange nodeVersion C.nodeVersionRange
         then do
-          let process = (P.proc command args) {P.cwd = Just $ SP.fromAbsDir fromDir}
+          let (specificCommand, specificArgs) = C.buildNpmCmdWithArgs command args
```

Comment:
Now that we renamed the function to `buildNpmCmdWithArgs`, and that it is used only for running `npm` + some args, I don't think we want to call it here anymore, since here we are dealing with any command that requires node installed, which can also be other stuff that is not npm, for example `npx`.

Reply:
Done)